tokenizers/tests/documentation.rs

```diff
@@ -451,10 +454,31 @@ fn pipeline_bert() -> tokenizers::Result<()> {
     use tokenizers::decoders::wordpiece::WordPiece as WordPieceDecoder;
 
     bert_tokenizer.with_decoder(WordPieceDecoder::default());
-    let decoded = bert_tokenizer.decode(output.get_ids(), true)?;
+    let decoded = bert_tokenizer.decode(output.get_ids(), true, false)?;
     // "welcome to the tokenizers library."
     // END bert_proper_decoding
     assert_eq!(decoded, "welcome to the tokenizers library.");
 
     Ok(())
 }
+
+#[test]
+fn spaces_between_added_tokens() -> tokenizers::Result<()> {
+    let mut bert_tokenizer = Tokenizer::from_file("data/bert-wiki.json")?;
+    bert_tokenizer.add_special_tokens(&[
+        AddedToken::from("[ABC]", false),
+        AddedToken::from("[DEF]", false),
+        AddedToken::from("GHI IHG", false),
+    ]);
+    let input_ids = bert_tokenizer.encode("[ABC][DEF][ABC]GHI IHG[DEF]", false)?;
+    let decoded_wo_spaces = bert_tokenizer.decode(input_ids.get_ids(), true, false)?;
+    println!("{}", decoded_wo_spaces);
+    assert_eq!(decoded_wo_spaces, "[ABC][DEF][ABC]GHI IHG[DEF]");
+
+    let decoded_w_spaces = bert_tokenizer.decode(input_ids.get_ids(), true, true)?;
+    println!("{}", decoded_w_spaces);
+    // "[ABC] [DEF] [ABC] GHI IHG [DEF]"
+    assert_ne!(decoded_wo_spaces, decoded_w_spaces);
+    assert_eq!(decoded_w_spaces, "[ABC] [DEF] [ABC] GHI IHG [DEF]");
```

Comment:
Do you mind adding a special in the mix so we can check that skip still works with spaces between ?

Reply:
Sure!